nav2_planner/src/planner_server.cpp

```diff
@@ -510,7 +516,52 @@ PlannerServer::publishPlan(const nav_msgs::msg::Path & path)
   }
 }
 
-}  // namespace nav2_planner
+void PlannerServer::isPathValid(
+  const std::shared_ptr<nav2_msgs::srv::IsPathValid::Request> request,
+  std::shared_ptr<nav2_msgs::srv::IsPathValid::Response> response)
+{
+  unsigned int mx = 0;
+  unsigned int my = 0;
+
+  response->is_valid = true;
+
+  if (request->path.poses.empty() ) {
+    response->is_valid = false;
+  } else {
+    geometry_msgs::msg::PoseStamped current_pose;
+    unsigned int closest_pose_index = 0;
+    if (costmap_ros_->getRobotPose(current_pose)) {
+      tf2::Vector3 curr_position(current_pose.pose.position.x,
+        current_pose.pose.position.y,
+        current_pose.pose.position.z);
+      // Find index of pose closest to the robot
+      float closest_distance = 0;
+      for (unsigned int i = 0; i < request->path.poses.size(); ++i) {
+
+        tf2::Vector3 path_point(request->path.poses[i].pose.position.x,
+          request->path.poses[i].pose.position.y,
+          request->path.poses[i].pose.position.z);
+
+        if (tf2::tf2Distance(curr_position, path_point) < closest_distance) {
+          closest_pose_index = i;
+          closest_distance = tf2::tf2Distance(curr_position, path_point);
+        }
+      }
+    }
+
+    for (unsigned int i = closest_pose_index; i < request->path.poses.size(); ++i) {
```

Comment:
Add a comment here that we're using the closest point to the end that way we don't check passed points if they're occupied now, because we passed them already so we don't care.

Reply:
Done